src/basic/rm-rf.c

```diff
@@ -187,6 +187,13 @@ int rm_rf(const char *path, RemoveFlags flags) {
                 return -EPERM;
         }
 
+        /* Another safe-check. Removing "/path/.." could easily remove entire root as well.
+         * It's especially easy to do using globs in tmpfiles, like "/path/.*", which the glob()
+         * function expands to both "/path/." and "/path/..".
+         * Return -EINVAL to be consistent with rmdir("/path/."). */
+        if (endswith(path, "/..") || endswith(path, "/../"))
+                return -EINVAL;
```

Comment:
I didn't really care about the normalization. I think that refusing to delete anything ending with ".." in addition to "." is just sane. The ```rm``` command does the same.